First commits

jsf9k opened the pull request:
## 🗣 Description ##

This pull request implements a Terraform module for installing into an account an AWS Lambda function that will disable users' inactive console and programmatic access.  The Lambda function is run on a schedule via AWS EventBridge.

See also:
- cisagov/provisionaccount-role-tf-module/pull/45
- cisagov/disable-inactive-iam-users-lambda#1
- cisagov/cool-accounts#147

## 💭 Motivation and context ##

We have a POAM about disabling inactive users in IAM.  Partially resolves cisagov/cool-system-internal#143.

## 🧪 Testing ##

I applied this module to our COOL Users account without incident.

## ✅ Pre-approval checklist ##

- [x] This PR has an informative and human-readable title.
- [x] Changes are limited to a single goal - *eschew scope creep!*
- [x] All relevant type-of-change labels have been added.
- [x] I have read the [CONTRIBUTING](../blob/develop/CONTRIBUTING.md) document.
- [x] These code changes follow [cisagov code standards](https://github.com/cisagov/development-guide).
- [x] All relevant repo and/or project documentation has been updated to reflect the changes in this PR.
- [x] All new and existing tests pass.

jsf9k committed:
First cut at adding functionality

jsf9k committed:
Rename file

The new name better reflects what the file contains.

jsf9k committed:
Make sure that iam:ListUsers permission has a resources block

It is an error not to have one.

jsf9k committed:
Correct schedule interval expression

Singular units are required for a value of one.  See
https://docs.aws.amazon.com/eventbridge/latest/userguide/eb-scheduled-rule-pattern.html#eb-rate-expressions
for more details.

jsf9k committed:
Add permission to retrieve the login profile for an IAM user

jsf9k committed:
Word-wrap two paragraphs to get rid of linting errors

jsf9k committed:
Update example code to work with this module

jsf9k committed:
Update Terraform documentation to match changes

jsf9k committed:
Remove blurb that only applies to skeletons

jsf9k committed:
Update introduction and usage example to match code in this repo

jsf9k committed:
Uncomment Dependabot ignore lines from upstream

Also correct repo reference in comments.

Co-authored-by: Nick <50747025+mcdonnnj@users.noreply.github.com>

dav3r commented on README.md:
```diff
@@ -52,28 +42,40 @@ No modules.
 | aws\_region | The AWS region to deploy into (e.g. us-east-1). | `string` | `"us-east-1"` | no |
-| subnet\_id | The ID of the AWS subnet to deploy into (e.g. subnet-0123456789abcdef0). | `string` | n/a | yes |
+| expiration\_days | A strictly positive integer denoting the number of days after which an IAM user's access is considered inactive if unused. | `number` | `45` | no |
+| lambda\_function\_description | The description of the Lambda function. | `string` | `"Lambda function to disable inactive IAM users."` | no |
```
I think we usually use the same verbiage in the description as we do in the name:
```suggestion
| lambda\_function\_description | The description of the Lambda function that will disable inactive IAM users. | `string` | `"Lambda function to disable inactive IAM users."` | no |
```

dav3r commented on variables.tf:
```diff
@@ -4,30 +4,67 @@
+
+variable "lambda_function_description" {
+  default     = "Lambda function to disable inactive IAM users."
+  description = "The description of the Lambda function."
```
Corresponding to a similar suggestion above:
```suggestion
  description = "The description of the Lambda function that will disable inactive IAM users."
```

dav3r commented on lambda.tf:
```diff
@@ -0,0 +1,49 @@
+  handler          = "lambda_handler.handler"
+  memory_size      = 128
+  role             = aws_iam_role.lambdaexecution_role.arn
+  runtime          = "python3.9"
```
Is there some reason why we wouldn't want to update this to run with `python3.11` or `python3.12`?

https://docs.aws.amazon.com/lambda/latest/dg/lambda-runtimes.html

jsf9k replied:
I was waiting for our skeleton to support it.  It would be easy enough to add, though.

dav3r replied:
In this case, I'm fine with getting ahead of the skeleton.  Let us know what you decide and I'll resolve this conversation either way.

jsf9k replied:
Our Terraform provider constraints only allow us to support Python 3.10 or earlier, so I will go with that.

jsf9k replied:
Please see commit 3ddb152ad5f212d25fae4204647e4327a78df46b and commit cisagov/disable-inactive-iam-users-lambda@311f8b8d2ea7997042970e4b1d025173f49ac28a.

dav3r replied:
Also, we need to move to a Terraform provider that supports Python 3.12 (required for what we are building over in https://github.com/cisagov/cyhy-kevsync-lambda/tree/first-commits-felddy).

dav3r reviewed (approved):
I noted a couple of thangs, but nothing earth-shattering. 👍

jsf9k committed:
Store the Lambda deployment package in a bucket

jsf9k committed:
Improve description of input variable

Co-authored-by: dav3r <david.redmin@trio.dhs.gov>

jsf9k committed:
Use the Python 3.10 runtime

This is the latest Pythin runtime we can currently support due to our
AWS Terraform provider constraints.

Co-authored-by: Dave Redmin <david.redmin@gwe.cisa.dhs.gov>

dav3r reviewed (approved):
👍 👍
